tests/cases/compiler/dynamicNames.ts

```diff
@@ -0,0 +1,109 @@
+// @target: esnext
+// @module: commonjs
+// @filename: module.ts
+export const c0 = "a";
+export const c1 = 1;
+export interface T0 {
+    [c0]: number;
+    [c1]: string;
+}
+export declare class T1 implements T2 {
+    [c0]: number;
+    [c1]: string;
+}
+export declare class T2 extends T1 {
+}
+export declare type T3 = {
+    [c0]: number;
+    [c1]: string;
+};
+
+// @filename: main.ts
+import { c0, c1, T0, T1, T2, T3 } from "./module";
+import * as M from "./module";
+
+namespace N {
+    export const c2 = "a";
+    export const c3 = 1;
+
+    export interface T4 {
+        [N.c2]: number;
+        [N.c3]: string;
+    }
+    export declare class T5 implements T4 {
+        [N.c2]: number;
+        [N.c3]: string;
+    }
+    export declare class T6 extends T5 {
+    }
+    export declare type T7 = {
+        [N.c2]: number;
+        [N.c3]: string;
+    };
+}
+
+const c4 = "a";
+const c5 = 1;
+const c6 = "1";
+
+interface T8 {
+    [c4]: number;
+    [c5]: string;
+}
+declare class T9 implements T8 {
+    [c4]: number;
+    [c5]: string;
+}
+declare class T10 extends T9 {
+}
+declare type T11 = {
+    [c4]: number;
+    [c5]: string;
+};
+
+interface T12 {
+    a: number;
+    1: string;
+}
+declare class T13 implements T2 {
+    a: number;
+    1: string;
+}
+declare class T14 extends T13 {
+}
+declare type T15 = {
+    a: number;
+    1: string;
+};
+
+interface T16 {
+    [c5]: number;
+    [c6]: string;

```

Comment:
Shouldn't this be an error? I also tested with duplicate string literals with incompatible types. The expected behaviour is a compile time error, however a union may also be possible.

```ts
const k1 = "literal name";
const k2 = "literal name";

interface T17 {
  [k1]: string;
  [k2]: 123;  // error expected
}

type T18 = {
  [k1]: string;
  [k2]: 123;  // error expected
}

// strange IntelliSense:
// type T18 = { literal name: string }

interface T19 {
  ['literal name']: string;
  ['literal name']: 123; // error as expected
}

interface T20 {
  ['literal name']: string; // error here
  [k1]: 123;                // and also here
}

let t17: T17;
t17['literal name'] // strange IntelliSense: `T17.literal name: string`

let t19: T19;
t19['literal name'] // strange IntelliSense: `T19[['literal name']]: string`
```